### ISSUE TITLE
Improvement: Add support for .NET libraries

### ISSUE THREAD

**Issue by milnet92** [milnet92/MXTXppInterpreter#49]
I'm working on adding support for .NET library references. It's under construction in [features/net_assemblies](/milnet92/MXTXppInterpreter/tree/feature/net_assemblies) branch. 

As of now, It's mature enough for the interpreter to interpret itself 🏃 

![image](https://github.com/user-attachments/assets/bf4e4390-84a0-4fd8-88c2-1e8d326097ee)

**Comment by milnet92** [comment]
Current limitations:
- Generic types cannot be used in declarations
- *use* keyword is not supported so types from .net references must be used with full type name (i.e. to use **namespace.type**)
- Optional parameters must be always passed
- Methods with multiple declarations can be used, but the editor will show always the first declaration

**Comment by bakarbsa** [comment]
So, can we upload and read excel file after this feature released? because we need some declaration like System.IO.Stream, OfficeOpenXml, etc. Thanks!!

**Comment by milnet92** [comment]
Yes you will be able to use any .NET reference you want that It can be used right now in F&O.

**Comment by TrudAX** [comment]
@milnet92  I faced a situation recently when I needed this. Do you think it can be merged with the main branch? (assume that it will not break existing functionality)

**Comment by vukisz** [comment]
> [@milnet92](https://github.com/milnet92) I faced a situation recently when I needed this. Do you think it can be merged with the main branch? (assume that it will not break existing functionality)

Same here quite often need to run helper classes and do full builds to test things which involves .net libs (extremely often of internal pricing ones)
